Added new React Guide for Authenticated Routes (#1675)
aws-amplify/amplify-ui: docs/src/pages/guides/guide.web.mdx

## Guides

Amplify guides are meant to give you a more in-depth understanding of how to use the Amplify UI components and theming to build out common functionality, end-to-end solutions, and frequently asked for workflows.

## Overview

Guides are organized by topic, but some guides may overlap multiple topics (i.e. Authentication & Theming) to demonstrate how to integrate multiple services together.

> Guides may combine the CLI and libraries together in a single tutorial. This is done to teach an end-to-end solution using both the service creation and configuration as well as connecting to and interacting with the service from a client-side application.

Reviewer: ```suggestion
Guides may combine the CLI and libraries together in a single tutorial. This is done to provide an end-to-end solution using both the service creation and configuration as well as connecting to and interacting with the service from a client-side application.
```

Author: 👍

Author: 7c6ee573f1e15b8407892dfd8552ea416e8cdf63

## Contributing

> If there is a guide or idea that you would like to see implemented that is not listed, open a GitHub Issue with the details of the use case and it will be reviewed by our team and considered.

Reviewer: ```suggestion
If there is a guide or idea that you would like to see implemented that is not listed, open a [GitHub Issue](https://github.com/aws-amplify/amplify-ui/issues/new/choose) with the details of the use case and it will be reviewed by our team and considered.
```

Author: 👍

Author: 7c6ee573f1e15b8407892dfd8552ea416e8cdf63

> If you have an idea for a guide that you'd like to write yourself you can submit a pull request. Before writing the guide and submitting the pull request, submit an issue with the details of your idea and you will receive guidance and feedback.

Reviewer: ```suggestion
If you have an idea for a guide that you'd like to write yourself and submit a pull request, submit an issue with the details of your idea before you submit your PR and you will receive guidance and feedback.
```

Author: I edited this line, with your suggestions and a few of my own changes to make it sound more natural.

Author: 7c6ee573f1e15b8407892dfd8552ea416e8cdf63

Reviewer: I like it!
